bot/StateRelayerBot.ts

```diff
@@ -36,109 +20,41 @@ export async function handler(props: StateRelayerHandlerProps): Promise<DFCData
+    const inputForDexUpdate = tranformPairData(rawPoolPairData, statsData, dexPriceData);
 
     // Data from vaults
-    const totalLoanValue = statsData.loan.value.loan;
-    const totalCollateralValue = statsData.loan.value.collateral;
-    dataVault.noOfVaults = transformToEthersBigNumber(statsData.loan.count.openVaults.toString(), 0);
-    dataVault.totalLoanValue = transformToEthersBigNumber(totalLoanValue.toString(), DECIMALS);
-    dataVault.totalCollateralValue = transformToEthersBigNumber(totalCollateralValue.toString(), DECIMALS);
-    dataVault.totalCollateralizationRatio = transformToEthersBigNumber(
-      ((totalCollateralValue / totalLoanValue) * 100).toFixed(3).toString(),
-      DECIMALS,
-    );
-    dataVault.activeAuctions = transformToEthersBigNumber(statsData.loan.count.openAuctions.toString(), 0);
-    dataVault.decimals = DECIMALS;
+    const dataVault = transformDataVault(statsData);
 
     // Data from Master Nodes
-    dataMasterNode.totalValueLockedInMasterNodes = transformToEthersBigNumber(
-      statsData.tvl.masternodes.toString(),
-      DECIMALS,
-    );
-    dataMasterNode.zeroYearLocked = transformToEthersBigNumber(statsData.masternodes.locked[0].count.toString(), 0);
-    dataMasterNode.fiveYearLocked = transformToEthersBigNumber(statsData.masternodes.locked[2].count.toString(), 0);
-    dataMasterNode.tenYearLocked = transformToEthersBigNumber(statsData.masternodes.locked[1].count.toString(), 0);
-    dataMasterNode.decimals = DECIMALS;
+    const dataMasterNode = transformDataMasternode(statsData);
 
-    // Get Data from all burns in ecosystem
-    burnedData.fee = transformToEthersBigNumber(statsData.burned.fee.toString(), DECIMALS);
-    burnedData.auction = transformToEthersBigNumber(statsData.burned.auction.toString(), DECIMALS);
-    burnedData.payback = transformToEthersBigNumber(statsData.burned.payback.toString(), DECIMALS);
-    burnedData.emission = transformToEthersBigNumber(statsData.burned.emission.toString(), DECIMALS);
-    burnedData.total = transformToEthersBigNumber(statsData.burned.total.toString(), DECIMALS);
-    burnedData.decimals = DECIMALS;
     // Call SC Function to update Data
     // Update Dex information
     const dexInfoTx = await stateRelayerContract.updateDEXInfo(
-      Object.keys(dataStore.pair),
-      Object.values(dataStore.pair),
-      totalValueLockInPoolPair,
-      total24HVolume,
+      inputForDexUpdate.dex,
+      inputForDexUpdate.dexInfo,
+      inputForDexUpdate.totalValueLocked.toString(),
+      inputForDexUpdate.total24HVolume.toString(),
     );
 
     // Update Master Node information
     const masterDataTx = await stateRelayerContract.updateMasterNodeInformation(dataMasterNode);
     // Update Vault general information
     const vaultTx = await stateRelayerContract.updateVaultGeneralInformation(dataVault);
-    // Update Burn information
-    const burnTx = await stateRelayerContract.updateBurnInfo(burnedData);
     if (!props.testGasCost) {
```

Comment:
Curious what is this for?

Reply:
In the case of production use, we don't need receipt for gas estimation purpose, so the function omit the receipts in return values